lib/config.go

```diff
@@ -2,65 +2,110 @@ package lib
 type RundeckConfig struct {
-	Url    string `json:"url"`
-	ApiKey string `json:"api_key"`
+	URL    string `json:"url"`
+	APIKey string `json:"api_key"`
 }
 
+// Config struct holds the setup config file info
 type Config struct {
 	Zabbix  ZabbixConfig  `json:"zabbix"`
 	Rundeck RundeckConfig `json:"rundeck"`
 }
 
 var (
-	UserHome        string
-	ConfigPath      string
+	// ConfigPath holds path to config
+	ConfigPath string
+
+	// ConfigDirectory holds path to config directory
 	ConfigDirectory string
 )
 
-const APP_CONFIG_DIRECTORY = "/.rundeck-zabbix/"
-const APP_CONFIG_NAME = "config"
-const APP_CONFIG_EXTENSION = "json"
+// AppConfigName is the config file constant for config
+const AppConfigName = "config"
+
+// ZabbixConfigKeyName is the config file constant for zabbix
+const ZabbixConfigKeyName = "zabbix"
+
+// RundeckConfigKeyName is the config file constant for rundeck
+const RundeckConfigKeyName = "rundeck"
+
+// URLConfigKeyName is the config file constant for url
+const URLConfigKeyName = "url"
+
+// APIConfigKeyName is the config file constant for api_key
+const APIConfigKeyName = "api_key"
+
+// UserConfigKeyName is the config file constant for username
+const UserConfigKeyName = "username"
+
+// PasswordConfigKeyName is the config file constant for password
+const PasswordConfigKeyName = "password"
 
 func init() {
-	UserHome = os.ExpandEnv("$HOME")
-	ConfigDirectory = UserHome + APP_CONFIG_DIRECTORY
-	ConfigPath = ConfigDirectory + APP_CONFIG_NAME + "." + APP_CONFIG_EXTENSION
+	ConfigDirectory = path.Join(os.ExpandEnv("$HOME"), ".rundeck-zabbix/")
+	ConfigPath = path.Join(ConfigDirectory, fmt.Sprintf("%s.json", AppConfigName))
+
 	if FileExists(ConfigDirectory) == false {
 		os.MkdirAll(ConfigDirectory, os.ModePerm)
 	}
 }
 
-func (object *Config) Save() {
+// Save object config
+func (object *Config) Save() error {
 	data, err := json.MarshalIndent(*object, "", "  ")
 	if err != nil {
-		panic(err)
+		return fmt.Errorf("cannot marshal config. Error: %v", err)
 	}
+
 	err = ioutil.WriteFile(ConfigPath, data, 0644)
 	if err != nil {
-		panic(err)
+		return fmt.Errorf("cannot write to file. Error: %v", err)
 	}
+	return nil
 }
 
-func NewConfigFromFile() *Config {
+// NewConfigFromFile creates new config from file
+func NewConfigFromFile() (*Config, error) {
 	config := Config{}
 	b, err := ioutil.ReadFile(ConfigPath)
 	if err != nil {
-		panic(err)
+		return nil, fmt.Errorf("cannot read file. Error: %v", err)
 	}
 	err = json.Unmarshal(b, &config)
 	if err != nil {
-		panic(err)
+		return nil, fmt.Errorf("cannot unmarshal json. Error: %v", err)
+	}
+	return &config, nil
+}
+
+// GetConfigValueByKey gets value from config file loaded via viper
+func GetConfigValueByKey(configKey, valueKey string) string {
+	k := fmt.Sprintf("%s.%s", configKey, valueKey)
+	return viper.GetString(k)
+}
+
+// FileExists checks if file exists and returns a bool value
+func FileExists(name string) bool {
```

Comment:
This seems to be used in `lib/config.go` only. Safe to have it be a private function.

Reply:
It is also used in cli/cmd/root.go